core/src/main/java/org/apache/any23/extractor/openie/OpenIEExtractor.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.any23.extractor.ExtractionContext;
+import org.apache.any23.extractor.ExtractorDescription;
+import org.apache.any23.util.StreamUtils;
+import org.eclipse.rdf4j.model.IRI;
+import org.apache.any23.extractor.ExtractionException;
+import org.apache.any23.extractor.ExtractionParameters;
+import org.apache.any23.extractor.ExtractionResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+
+import edu.knowitall.openie.Argument;
+import edu.knowitall.openie.Instance;
+import edu.knowitall.openie.OpenIE;
+import edu.knowitall.tool.parse.ClearParser;
+import edu.knowitall.tool.postag.ClearPostagger;
+import edu.knowitall.tool.srl.ClearSrl;
+import edu.knowitall.tool.tokenize.ClearTokenizer;
+import scala.collection.JavaConversions;
+import scala.collection.Seq;
+
+
+
+/**
+ * An <a href="https://github.com/allenai/openie-standalone">OpenIE</a> 
+ * extractor able to generate <i>RDF</i> statements from 
+ * sentences representing relations in the text.
+ */
+public class OpenIEExtractor implements Extractor.TagSoupDOMExtractor {
+
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
+
+    private IRI documentRoot;
+
+    /**
+     * default constructor
+     */
+    OpenIEExtractor() {
+        // default constructor
+    }
+
+    /**
+     * @see org.apache.any23.extractor.Extractor#getDescription()
+     */
+    @Override
+    public ExtractorDescription getDescription() {
+        return OpenIEExtractorFactory.getDescriptionInstance();
+    }
+
+    @Override
+    public void run(ExtractionParameters extractionParameters,
+            ExtractionContext context, Document in, ExtractionResult out)
+                    throws IOException, ExtractionException {
+        OpenIE openIE = new OpenIE(new ClearParser(new ClearPostagger(new ClearTokenizer())), new ClearSrl(), false, false);
+
+
+        Seq<Instance> extractions = null;
+        try {
+            extractions = openIE.extract(StreamUtils.asString(StreamUtils.documentToInputStream(in)));
+        } catch (TransformerConfigurationException | TransformerFactoryConfigurationError e) {
+            LOG.error("Error during extraction: {}", e);
+        }
+
+        List<Instance> listExtractions = JavaConversions.seqAsJavaList(extractions);
+        for(Instance instance : listExtractions) {
+            StringBuilder sb = new StringBuilder();
+
+            sb.append(instance.confidence())
+            .append('\t')
+            .append(instance.extr().context())
+            .append('\t')
+            .append(instance.extr().arg1().text())
+            .append('\t')
+            .append(instance.extr().rel().text())
+            .append('\t');
+
+            List<Argument> listArg2s = JavaConversions.seqAsJavaList(instance.extr().arg2s());
+            for(Argument argument : listArg2s) {
+                sb.append(argument.text()).append("; ");
+            }
+            System.out.println(sb.toString());
```

Comment:
Presuming this is stub code right now that will eventually send results to the ExtractorResult.